### PR TITLE
Add CI infrastructure

### PR DESCRIPTION
**Pull request by apmasell**
Copy configuration from numba-scipy

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -58,47 +57,61 @@ jobs:
...
 
   test_pyver:
-    runs-on: ubuntu-latest
     needs: build
     strategy:
       matrix:
-        pyver:
-          - 3.7
-          - 3.8
+        pyver: [3.6, 3.7, 3.8]
```
Comment (sklam) [inline .github/workflows/test_and_deploy.yml]: Latest numba doesn't support 3.6. And let's add new python versions
```suggestion
        pyver: [3.7, 3.8, 3.9, 3.10]
```

File: .github/workflows/test_and_deploy.yml [hunk 2]
```diff
@@ -58,47 +57,61 @@ jobs:
...
         run: |
           source "$CONDA/etc/profile.d/conda.sh"
-          conda activate ci
-          pytest --pyargs numba_extras
+          conda config --set always_yes yes --set changeps1 no
+          conda install -q anaconda-client
+          ls -la ./artifact_storage
+          UPLOAD=$(ls ./artifact_storage | head -1)
+          echo "Uploading $UPLOAD with label=${{ env.AC_LABEL }}"
+          $CONDA/bin/anaconda -t ${{ secrets.ANACONDA_ORG_TOKEN }} upload -u numba -l ${{ env.AC_LABEL }} --no-progress --force --no-register ./artifact_storage/$UPLOAD
```
Comment (sklam) [inline .github/workflows/test_and_deploy.yml:117]: i've added the secret token

File: .github/workflows/deploy_release.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          conda install -c numba python=3.8 numba>=0.45 wheel
+      - name: Build PyPI artifacts
+        shell: bash -l {0}
+        run: |
+          python setup.py sdist bdist_wheel
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.4.2
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_SECRET }}
```
Comment (sklam) [inline .github/workflows/deploy_release.yml]: Not sure if we should publish to PYPI just yet. We should probably see how it works on anaconda.org first.

File: .github/workflows/test_and_deploy.yml [hunk 4]
```diff
@@ -58,47 +57,61 @@ jobs:
...
           conda list
-      - name: numba system check
-        run: |
-          source "$CONDA/etc/profile.d/conda.sh"
-          conda activate ci
-          numba -s
       - name: Pytest
+        shell: bash -l {0}
+        run: |
+          pytest --pyargs numba_extras.tests
```
Comment (sklam) [inline .github/workflows/test_and_deploy.yml]: Testing for this repo is a bit different. Each submodule will be self-contained.

```suggestion
          pytest --pyargs numba_extras
```

File: .github/workflows/test_and_deploy.yml (hunk 2 above)
Comment (apmasell) [reply .github/workflows/test_and_deploy.yml:117]: Thank you.

**Comment by sklam** [comment]
going to merge it to see if deploy actually works

**Comment by esc** [comment]
@sklam how did it turn out?

**Pull request merged**